client/state/themes/selectors.js

```diff
@@ -357,7 +357,7 @@ export function getThemeDetailsUrl( state, theme, siteId ) {
  * @return {?String}        Theme setup instructions URL
  */
 export function getThemeSupportUrl( state, theme, siteId ) {
-	if ( ! theme || ! isPremiumTheme( theme ) ) {
+	if ( isJetpackSite( state, siteId ) || ! theme || ! isThemePremium( state, theme.id ) ) {
```

Comment:
I think I'd like to leave it in for now and revisit once we've settled what to do about that 'Setup' tab.